src/main/kotlin/io/kni/thingoo/backend/mqtt/MqttPubSubClientImpl.kt

```diff
@@ -53,8 +58,33 @@ class MqttPubSubClientImpl(
 
     private fun connectAsync() {
         val options = createMqttConnectOptions()
-        val token = client!!.connect(options)
-        token.waitForCompletion()
+        tryConnectAsyncWithRetries(options)
+    }
+
+    private fun tryConnectAsyncWithRetries(options: MqttConnectOptions) {
+        var retries = 0
+        while (true) {
+            if (retries == config.connectRetryCount) {
+                logger.error("[MQTT] Couldn't connect to broker in $retries retries. Closing Spring application.")
+                ctx.close()
+                exitProcess(-1)
+            }
+
+            val token = client!!.connect(options)
+            val timeoutInMillis = TimeUnit.SECONDS.toMillis(config.connectTimeout)
+            try {
+                retries++
+                logger.warn("[MQTT] Trying to connect to broker")
```

Comment:
should this really be at the `WARN` level?

Reply:
Good catch, resolved in https://github.com/ThingooKNI/backend/pull/79/commits/db7479ca48a65d0ce6c97d38da0857f0e340d05d